operations/helm/charts/mimir-distributed/templates/lib/securityContext.tpl

```diff
@@ -0,0 +1,12 @@
+{{/*
+Mimir common podSecurityContext
+Params:
+  ctx = . context
+  component = name of the component
+*/}}
+{{- define "mimir.lib.podSecurityContext" -}}
+{{- $componentSection := include "mimir.componentSectionFromName" . | fromYaml }}
+{{- /* Some components have security context in the values.yalm as podSecurityContext and others as securityContext */}}
+{{- $podSecurityContextSection := $componentSection.securityContext | default $componentSection.podSecurityContext -}}
+{{- $podSecurityContextSection | mustMergeOverwrite .ctx.Values.rbac.podSecurityContext | toYaml -}}
```

Comment:
This actually changes the value of `.ctx.Values.rbac.podSecurityContext`. So changing the security context of one component ends up changing the security contexts of all components because that one components modifies the global `.ctx.Values.rbac.podSecurityContext`

I think doing something like this should fix it

```suggestion 
{{- mustMergeOverwrite (dict) $podSecurityContextSection .ctx.Values.rbac.podSecurityContext  | toYaml -}}
```


Also, isn't the order in the current version wrong - shouldn't component sections take precedence over the global `rbac` section?

Reply:
I've added a deepCopy to stop overwriting the values.
The order is correct, just tested it.